app/controllers/courses_controller.rb

```diff
@@ -1,5 +1,5 @@
 class CoursesController < ApplicationController
   def index
-    # @courses = current_user.courses
+    @courses = current_user.courses.includes(:roles)
```

Comment:
This generates the following:

    SELECT `roles`.* FROM `roles` WHERE `roles`.`course_id` IN (2, 3)

This will load all roles for those courses, which includes all other students, markers, instructors for that course.

This might look a bit incorrect from a human perspective, but the correct way to do this is:

    current_use.roles.includes(:course)